packages/flutter_tools/lib/src/commands/symbolicate.dart

```diff
@@ -0,0 +1,125 @@
+    );
+    argParser.addOption(
+      'input',
+      abbr: 'i',
+      valueHelp: '/crashes/stack_trace.err',
+      help: 'A file path containing a Dart stack trace. Only supported when '
+        'paired with the "--output" option.'
+    );
+    argParser.addOption(
+      'output',
+      abbr: 'o',
+      valueHelp: 'A file path for a symbolicated stack trace to be written to. '
+        'Only supported when paired with the "--input" option.'
+    );
+  }
+
+  final Stdio _stdio;
+  final FileSystem _fileSystem;
+
+  @override
+  String get description => 'symbolicate a stack trace from an AOT compiled flutter application.';
+
+  @override
+  String get name => 'symbolicate';
+
+  @override
+  bool get shouldUpdateCache => false;
+
+  @override
+  Future<void> validateCommand() {
+    if (!argResults.wasParsed('debug-info')) {
+      throwToolExit('"--debug-info" is required to symbolicate stack traces.');
+    }
+    if (!_fileSystem.isFileSync(stringArg('debug-info'))) {
+      throwToolExit('${stringArg('debug-info')} does not exist.');
+    }
+    if ((argResults.wasParsed('input') && !argResults.wasParsed('output')) ||
+        (!argResults.wasParsed('input') && argResults.wasParsed('output'))) {
+      throwToolExit('"--input" and "--output" are only supported when both are provided.');
+    }
+    if (argResults.wasParsed('input') && !_fileSystem.isFileSync(stringArg('input'))) {
+      throwToolExit('${stringArg('input')} does not exist.');
+    }
+    return super.validateCommand();
+  }
+
+  // Validated above that both need to be provided.
+  bool get _usingFileValues => argResults.wasParsed('input') && argResults.wasParsed('output');
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    _stdio.stdin.transform(const Utf8Decoder()).listen(print, onDone: () {print('Done');});
+    Stream<List<int>> input;
+    IOSink output;
+
+    if (_usingFileValues) {
+      final File inputFile = _fileSystem.file(stringArg('input'));
+      final File outputFile = _fileSystem.file(stringArg('output'));
+      if (!outputFile.parent.existsSync()) {
+        outputFile.parent.createSync(recursive: true);
+      }
+      input = inputFile.openRead();
+      outputFile.createSync();
+      output = outputFile.openWrite();
+    } else {
+      input = _stdio.stdin;
+      output = _stdio.stdout;
+    }
+
+    final Uint8List debugInfo = _fileSystem.file(stringArg('debug-info')).readAsBytesSync();
+    final Dwarf dwarf = Dwarf.fromElf(Elf.fromReader(Reader.fromTypedData(debugInfo)));
+    if (dwarf == null) {
+      throwToolExit('Failed to decode debug file at ${stringArg('debug-info')}');
+    }
+
+    final Completer<void> onDone = Completer<void>();
+    input
+      .transform(const Utf8Decoder())
+      .transform(const LineSplitter())
+      .transform(DwarfStackTraceDecoder(dwarf, includeInternalFrames: true))
+      .listen(output.writeln, onDone: onDone.complete, onError: onDone.completeError);
```

Comment:
I would prefer folks not have to say `flutter run | flutter symbolize`. `flutter run` should take a flag to point at debugging info, or look around the project for one that matches the targeted device.

Reply:
Yeah I agree, I should wire up something to run - though I can imagine us not applying dwarf-stack-traces in run release too.